src/jitlayers.cpp

```diff
@@ -1086,7 +1088,7 @@ uint64_t JuliaOJIT::getFunctionAddress(StringRef Name)
     return cantFail(addr.getAddress());
 }
 
-static int globalUniqueGeneratedNames;
+static std::atomic<int> globalUniqueGeneratedNames;
 StringRef JuliaOJIT::getFunctionAtAddress(uint64_t Addr, jl_code_instance_t *codeinst)
```

Comment:
This function is not safe to call from multiple threads, so making the single variable atomic is probably not actually very helpful.